### PR TITLE
[compiler] implement rng uniform and normal

### PR DESCRIPTION
**Pull request by Mi-Jiazhi**

### CODE REVIEW

File: compiler/lib/Dialect/mhlo/Transforms/HloAggressiveFusion.cpp
Comment (qingyunqu) [inline compiler/lib/Dialect/mhlo/Transforms/HloAggressiveFusion.cpp:1]: should remove this file?

File: compiler/doc/rng.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+## RNG Normal Converter
+
+We use Box–Muller transform and Philox4×32-10 to implemet Normal Distribution Generator. Box–Muller transform takes two samples from the uniform distribution on the interval (0,1) and maps them to two standard, normally distributed samples.
+So We use the Philox4×32-10 and Normalization to generate two uniformly distributed random numbers whose range from 0 to 1 (just like the Uniform Distribution Generator, but this time we need to take two of the four Philox4×32-10 outputs and discard the remaining two) and then use Box–Muller transform to map them to standard, normally distributed values.
+So to generate a normal distributed random numbers, We need to perform a philox4×32-10 and an Box-Muller transform once. But if we can allow multiple random numbers to be generated at once, since a philox4×32-10 can generate 4 uniformly distributed random numbers, we can generate 6 pairs of inputs to the Box-Muller transform and generate 12 normal distributed random numbers with one philox4×32-10. It can greatly reduce the number of executions of philox4×32-10.
```
Comment (qingyunqu) [inline compiler/doc/rng.md:12]: Could you add `hlo-to-linalg` lowering case to show details? Also add math descriptions?

Comment (qingyunqu) [reply compiler/doc/rng.md:12]: And add future work which could provide optimization opportunites.

File: compiler/lib/Dialect/mhlo/Transforms/HloAggressiveFusion.cpp
Comment (Mi-Jiazhi) [reply compiler/lib/Dialect/mhlo/Transforms/HloAggressiveFusion.cpp:1]: yes, thank you for careful review

File: compiler/doc/rng.md (hunk 1 above)
Comment (Mi-Jiazhi) [reply compiler/doc/rng.md:12]: Sure!

File: compiler/doc/rng.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+$$y_2=\sin(2 \pi x_1)*\sqrt{(-2\ln x_2)}$$
+So fisrt of all, We use the Philox4×32-10 and Normalization to generate two uniformly distributed random numbers whose range from 0 to 1 (just like the Uniform Distribution Generator, but this time we need to take two of the four Philox4×32-10 outputs and discard the remaining two). And then use Box–Muller transform to map them to standard, normally distributed values. The process of lowing the rng.Normal from MHLO to linalg will be shown below.
+
+### Future Work
+Just like Rng.Uniform converter, we generate a single random number during each iteration. So for generating a normal distributed random numbers, We need to perform a philox4×32-10 and an Box-Muller transform once. But if we can allow multiple random numbers to be generated during each iteration, since one execution of philox4×32-10 can generate 4 uniformly distributed random numbers, we can generate 6 pairs of inputs to the Box-Muller transform and generate 12 normal distributed random numbers with one philox4×32-10. It can greatly reduce the number of executions of philox4×32-10.
+
+### Example of Lowing the Rng.Normal from MHLO to Linalg
```
Comment (qingyunqu) [inline compiler/doc/rng.md:20]: Also adding an example and math description of `rng.uniform`?

**Review by qingyunqu — APPROVED** [review]
LGTM, thanks for the efforts.

**Pull request merged**